x-pack/plugins/apm/server/routes/metrics/by_agent/shared/memory/index.ts

```diff
@@ -169,8 +193,12 @@ export async function getMemoryChartData({
             memoryUsedMax: { max: { script: percentSystemMemoryUsedScript } },
           },
           additionalFilters: [
-            { exists: { field: METRIC_SYSTEM_FREE_MEMORY } },
-            { exists: { field: METRIC_SYSTEM_TOTAL_MEMORY } },
+            {
+              bool: {
+                should: [systemMemoryFilter],
+                minimum_should_match: 1,
+              },
+            },
```

Comment:
I don't think this should be changed to a `should` clause. Both fields are required afaict

Reply:
good catch!

Reply:
Did any of our API test catch this? If not, can you add one that will?

Reply:
(thinking about this again: since the script already does a check and returns `null` there might not be anything to catch)